### PR TITLE
Docker and dependencies

### PR DESCRIPTION
**Pull request by ph1ne4s**
Did some changes in the Docker File and version of dependencies.

## Description
Python3.9 couldn't be installed on Kali: latest, so changed the docker image to ubuntu:20.04, And also changed some versions for easy installation.

## Related Issue
Resolves issue #1215 , #1213 

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [x ] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM kalilinux/kali-rolling:latest
```
Comment (viyatb) [inline docker/Dockerfile]: Moving away from Kali linux will break a lot of tools and plugins that are included with OWTF. I would recommend you keep this unchanged.

File: docker/Dockerfile [hunk 2]
```diff
@@ -30,9 +30,9 @@ ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
```
Comment (viyatb) [inline docker/Dockerfile]: ditto

**Review by viyatb — COMMENTED** [review]
Mostly ok - look at https://gist.github.com/jprjr/7667947. There are ways to install Python 3.9 even on the latest Ubuntu or Kali linux version without building it from the source. Pyenv is one such tool. 

Also submodule for `owtf/webapp/src/containers/Report/templates` should be removed.

**Comment by ph1ne4s** [comment]
@viyatb I have done the required changes, can you review?
Also I can't get the submodule as `webapp/src/containers/Report/templates` was empty.

File: docker/Dockerfile [hunk 3]
```diff
@@ -22,17 +22,39 @@ RUN apt-get -y update && apt-get clean
 RUN apt-get -y install  xvfb xserver-xephyr libxml2-dev libxslt-dev libssl-dev zlib1g-dev gcc python-all-dev \
 postgresql-server-dev-all postgresql-client postgresql-client-common \
 postgresql libcurl4-openssl-dev proxychains tor ca-certificates libpq-dev  \
-libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all \
-libyaml-dev python3.9 python3.9-dev build-essential python3.9-venv python3-lxml python3-impacket golang-go python3-pip
+libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all 
+#libyaml-dev python3.9 python3.9-dev build-essential python3.9-venv python3-lxml python3-impacket golang-go python3-pip
+# RUN apt-get install -y make build-essential libssl-dev git zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev
```
Comment (viyatb) [inline docker/Dockerfile]: remove commented code

File: docker/Dockerfile [hunk 4]
```diff
@@ -22,17 +22,39 @@ RUN apt-get -y update && apt-get clean
 RUN apt-get -y install  xvfb xserver-xephyr libxml2-dev libxslt-dev libssl-dev zlib1g-dev gcc python-all-dev \
 postgresql-server-dev-all postgresql-client postgresql-client-common \
 postgresql libcurl4-openssl-dev proxychains tor ca-certificates libpq-dev  \
-libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all \
-libyaml-dev python3.9 python3.9-dev build-essential python3.9-venv python3-lxml python3-impacket golang-go python3-pip
+libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all 
+#libyaml-dev python3.9 python3.9-dev build-essential python3.9-venv python3-lxml python3-impacket golang-go python3-pip
```
Comment (viyatb) [inline docker/Dockerfile]: can you remove the commented code?

File: docker/Dockerfile [hunk 5]
```diff
@@ -22,17 +22,39 @@ RUN apt-get -y update && apt-get clean
...
+# RUN apt-get install -y make build-essential libssl-dev git zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev
+# RUN curl https://pyenv.run | bash
+# RUN echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
+# RUN echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
+# RUN echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init --path)"\nfi' >> ~/.bashrc
```
Comment (viyatb) [inline docker/Dockerfile]: ditto, remove commented code.

File: docker/Dockerfile [hunk 6]
```diff
@@ -22,17 +22,39 @@ RUN apt-get -y update && apt-get clean
...
 
 # Set locale
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
 
-# Install optional tools (LBD, arachni, gnutls-bin, o-saft and metagoofil)
-RUN apt-get -y install lbd theharvester tlssled nikto dnsrecon nmap whatweb skipfish dirbuster \
-metasploit-framework wpscan wapiti hydra metagoofil o-saft amass
+# Install optional tools (LBD, arachni, gnutls-bin, o-saft and metagoofil) (not available on ubuntu20)
```
Comment (viyatb) [inline docker/Dockerfile]: without these tools, some of OWTF functionality will not be available - how can we use the same tools in Ubuntu, any ideas?

**Review by viyatb — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 6 above)
Comment (ph1ne4s) [reply docker/Dockerfile:48]: we changed to kali, so its working

Comment (ph1ne4s) [reply docker/Dockerfile]: yeah they can be installed as we are again using kali.
But sometimes gets this error at the end of docker build
image

Comment (viyatb) [reply docker/Dockerfile]: I think that's related to your machine running out of memory while building the image :) https://github.com/docker/cli/issues/2971

**Review by viyatb — APPROVED** [review]

File: docker/Dockerfile [hunk 7]
```diff
@@ -32,7 +47,7 @@ ENV LC_ALL en_US.UTF-8
 
 # Install optional tools (LBD, arachni, gnutls-bin, o-saft and metagoofil)
 RUN apt-get -y install lbd theharvester tlssled nikto dnsrecon nmap whatweb skipfish dirbuster \
-metasploit-framework wpscan wapiti hydra metagoofil o-saft amass
+metasploit-framework wpscan wapiti hydra metagoofil o-saft amass
```
Comment (viyatb) [inline docker/Dockerfile]: can you remove this trailing whitespace?

File: scripts/tools.yaml [hunk 8]
```diff
@@ -17,13 +17,13 @@ Panoptic:
...
 SSL-Cipher-check:
-  command: "wget --user-agent='Mozilla/5.0 (X11; Linux i686; rv:6.0) Gecko/20100101 Firefox/15.0' --tries=3 http://unspecific.com/ssl/ssl-cipher-check.pl; chmod 700 *"
+  command: "wget --user-agent='Mozilla/5.0 (X11; Linux i686; rv:6.0) Gecko/20100101 Firefox/15.0' --tries=3 https://github.com/philwo/pysk/blob/master/tools/ssl-cipher-check.pl; chmod 700 *"
   directory: tools/restricted/ssl/ssl-cipher-check
 SSLLABS:
   command: "wget --tries=3 https://github.com/ssllabs/ssllabs-scan/releases/download/v1.4.0/ssllabs-scan_1.4.0-linux64.tgz; tar xvf * ; rm *.tgz"
   directory: tools/restricted/ssllabs-scan
 SVN-Digger:
-  command: "wget --user-agent='Mozilla/5.0 (X11; Linux i686; rv:6.0) Gecko/20100101 Firefox/15.0' --tries=3 https://www.netsparker.com/s/research/SVNDigger.zip; unzip *.zip; rm -f *.zip"
+  command: "wget --user-agent='Mozilla/5.0 (X11; Linux i686; rv:6.0) Gecko/20100101 Firefox/15.0' --tries=3 https://github.com/ph1ne4s/SVNDigger/raw/main/SVNDigger.zip; unzip *.zip; rm -f *.zip"
```
Comment (viyatb) [inline scripts/tools.yaml]: why is this fetching SVNDigger from your fork `https://github.com/ph1ne4s/SVNDigger/raw/main`?

Comment (ph1ne4s) [reply scripts/tools.yaml]: The original source of SVNDigger was unavailable which was - https://www.invicti.com/blog/web-security/svn-digger-better-lists-for-forced-browsing/
(as netsparker changed to invicti)
so, I found its files from another repo (https://github.com/samduy/SVNDigger).
Should I change the source and what source should we use?

Comment (viyatb) [reply scripts/tools.yaml]: Ok use https://github.com/owtf/SVNDigger

**Review by viyatb — CHANGES_REQUESTED** [review]

**Pull request merged**